### ISSUE TITLE
Streaming support

### ISSUE THREAD

**Issue by liveforeverx** [devinus/poison#50]
Did you plan to add streaming support to `poison`?

**Comment by devinus** [comment]
Yes.

**Comment by teamon** [comment]
Just curious @liveforeverx, what more do you need besides:

```ex
stream |> Stream.map(&Poison.decode!/1)
```

? JSON stream is usually one document per line (with all "\n" escaped)

**Comment by sickill** [comment]
@liveforeverx is this about streaming parsing?

**Comment by isaacsanders** [comment]
Another way to interpret this is streaming a large dataset into the encoder and emitting pieces as they are available, as iodata.

**Comment by sickill** [comment]
Yeah, streamed encoding and streamed decoding are two different things, and both would be useful in specific situations.
